contrib/confluentinc/confluent-kafka-go/kafka/kafka.go

```diff
@@ -40,23 +41,41 @@ func NewProducer(conf *kafka.ConfigMap, opts ...Option) (*Producer, error) {
 	if err != nil {
 		return nil, err
 	}
-	return WrapProducer(p, opts...), nil
+
+	return WrapProducerWithConfig(p, conf, opts...), nil
 }
 
 // A Consumer wraps a kafka.Consumer.
 type Consumer struct {
 	*kafka.Consumer
-	cfg    *config
-	events chan kafka.Event
-	prev   ddtrace.Span
+	*kafka.ConfigMap
```

Comment:
```suggestion
	kafkaConfig *kafka.ConfigMap
```

It's better to not embed this struct and keep it private under the `kafkaConfig` namespace.

Also I think we could probably create a separate struct with only the fields we are interested in so we can save some memory.